File: OO_design_patterns_training_2022-11/Exercises/C#/Exercise/TextReporter.cs
using System;
using System.Collections.Generic;

namespace TestApp1
{
    public class TextReporter
    {
        public string OutputDir { get; set; }

        public void CreateDataTextFile(List<string> data)
        {
            Console.WriteLine("Data written as text file:");
            foreach (var item in data)
            {
                Console.WriteLine(item);
            }
        }

        public void CreateFooterTextFile()
        {
            Console.WriteLine($"Wrote an imaginary footer as a text file to {OutputDir}");
        }

        public void CreateHeaderTextFile()
        {
            Console.WriteLine($"Wrote an imaginary header as a text file to {OutputDir}");
        }
    }
}

File: OO_design_patterns_training_2022-11/Exercises/C++/Exercise/src/TextReporter.cpp
#include "TextReporter.h"

#include <iostream>

void TextReporter::CreateDataTextFile(const std::vector<std::string>& data) const
{
    std::cout << "Data written as text file:" << std::endl;
    for (const auto& item : data)
    {
        std::cout << item << std::endl;
    }
}

void TextReporter::CreateFooterTextFile() const
{
    std::cout << "Wrote an imaginary footer as a text file to " << _outputDir << std::endl;
}

void TextReporter::CreateHeaderTextFile() const
{
    std::cout << "Wrote an imaginary header as a text file to " << _outputDir << std::endl;
}
